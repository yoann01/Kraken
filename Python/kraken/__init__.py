"""Kraken Framework."""

import logging
import os

__all__ = ['core', 'helpers', 'plugins', 'ui']


krakenPath = os.path.abspath(os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..'))
if os.environ.get('KRAKEN_PATH', None) is None:
    os.environ['KRAKEN_PATH'] = krakenPath

krakenExtsPath = os.path.join(krakenPath, 'Exts')
if 'FABRIC_EXTS_PATH' in os.environ:
    if krakenExtsPath not in os.environ['FABRIC_EXTS_PATH']:
        os.environ['FABRIC_EXTS_PATH'] = krakenExtsPath + os.pathsep + os.environ['FABRIC_EXTS_PATH']

krakenPresetsPath = os.path.join(krakenPath, 'Presets', 'DFG')
if 'FABRIC_DFG_PATH' in os.environ:
    if krakenPresetsPath not in os.environ['FABRIC_DFG_PATH']:
        os.environ['FABRIC_DFG_PATH'] = krakenPresetsPath + os.pathsep + os.environ['FABRIC_DFG_PATH']


logging.basicConfig(format='[KRAKEN] %(levelname)s: %(message)s', level=logging.INFO)

# Custom inform level for use with UI label getting added to the status bar.
logging.INFORM = 25
logging.addLevelName(logging.INFORM, 'INFORM')
logging.Logger.inform = lambda inst, msg, *args, **kwargs: inst.log(logging.INFORM, msg, *args, **kwargs)


def release():
    print "RELEASE THE KRAKEN!!!!"
