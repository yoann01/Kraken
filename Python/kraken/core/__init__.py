"""Kraken Core."""

VERSION_MAJOR = 1
VERSION_MINOR = 2
VERSION_BUILD = 6
VERSION_SUFFIX = ""


def getVersion():
    """Contatenates the version globals and returns the current version of
    Kraken.

    Returns:
        str: Current version of Kraken.

    """

    versionString = str(VERSION_MAJOR) + "." + str(VERSION_MINOR) + "." + str(VERSION_BUILD)
    if VERSION_SUFFIX:
        versionString = versionString + "-" + VERSION_SUFFIX

    return versionString
