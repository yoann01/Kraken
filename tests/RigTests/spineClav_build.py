import json

from kraken import plugins
from kraken_examples.spineClav_rig import SpineClavRig
from kraken.core.profiler import Profiler
from kraken.helpers.utility_methods import logHierarchy


Profiler.getInstance().push("spineClav_build")

spineClavRig = SpineClavRig("char_bob")

builder = plugins.getBuilder()
builder.build(spineClavRig)

Profiler.getInstance().pop()

if __name__ == "__main__":
    print Profiler.getInstance().generateReport()
else:
   logHierarchy(spineClavRig)
