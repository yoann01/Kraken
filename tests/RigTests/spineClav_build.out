Failed to find DCC builder. Falling back to Python builder.
buildContainer: char_bob as: char_bob
building: char_bob.deformers as: char_bob_deformers type: Layer
buildLayer: char_bob.deformers as: char_bob_deformers
building: char_bob.deformers.spine as: spine_M_cmp type: ComponentGroup
buildGroup: char_bob.deformers.spine as: spine_M_cmp
building: char_bob.deformers.spine.spine01 as: spine_M_spine01_def type: Joint
buildJoint: char_bob.deformers.spine.spine01 as: spine_M_spine01_def
building: char_bob.deformers.spine.pelvis as: spine_M_pelvis_def type: Joint
buildJoint: char_bob.deformers.spine.pelvis as: spine_M_pelvis_def
building: char_bob.deformers.spine.spine02 as: spine_M_spine02_def type: Joint
buildJoint: char_bob.deformers.spine.spine02 as: spine_M_spine02_def
building: char_bob.deformers.spine.spine03 as: spine_M_spine03_def type: Joint
buildJoint: char_bob.deformers.spine.spine03 as: spine_M_spine03_def
building: char_bob.deformers.spine.spine04 as: spine_M_spine04_def type: Joint
buildJoint: char_bob.deformers.spine.spine04 as: spine_M_spine04_def
building: char_bob.deformers.clavicleGuide as: clavicleGuide_L_cmp type: ComponentGroup
buildGroup: char_bob.deformers.clavicleGuide as: clavicleGuide_L_cmp
building: char_bob.deformers.clavicleGuide.clavicle as: clavicleGuide_L_clavicle_def type: Joint
buildJoint: char_bob.deformers.clavicleGuide.clavicle as: clavicleGuide_L_clavicle_def
building: char_bob.controls as: char_bob_controls type: Layer
buildLayer: char_bob.controls as: char_bob_controls
building: char_bob.controls.spine as: spine_M_cmp type: ComponentGroup
buildGroup: char_bob.controls.spine as: spine_M_cmp
building: char_bob.controls.spine.inputs as: spine_M_inputs_hrc type: HierarchyGroup
buildHierarchyGroup: char_bob.controls.spine.inputs as: spine_M_inputs_hrc
building: char_bob.controls.spine.inputs.mainSrt as: spine_M_mainSrt_cmpIn type: ComponentInput
buildLocator: char_bob.controls.spine.inputs.mainSrt as: spine_M_mainSrt_cmpIn
building: char_bob.controls.spine.outputs as: spine_M_outputs_hrc type: HierarchyGroup
buildHierarchyGroup: char_bob.controls.spine.outputs as: spine_M_outputs_hrc
building: char_bob.controls.spine.outputs.cog as: spine_M_cog_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.cog as: spine_M_cog_cmpOut
building: char_bob.controls.spine.outputs.spineBase as: spine_M_spineBase_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spineBase as: spine_M_spineBase_cmpOut
building: char_bob.controls.spine.outputs.pelvis as: spine_M_pelvis_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.pelvis as: spine_M_pelvis_cmpOut
building: char_bob.controls.spine.outputs.spineEnd as: spine_M_spineEnd_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spineEnd as: spine_M_spineEnd_cmpOut
building: char_bob.controls.spine.outputs.spine01 as: spine_M_spine01_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spine01 as: spine_M_spine01_cmpOut
building: char_bob.controls.spine.outputs.spine02 as: spine_M_spine02_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spine02 as: spine_M_spine02_cmpOut
building: char_bob.controls.spine.outputs.spine03 as: spine_M_spine03_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spine03 as: spine_M_spine03_cmpOut
building: char_bob.controls.spine.outputs.spine04 as: spine_M_spine04_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spine04 as: spine_M_spine04_cmpOut
building: char_bob.controls.spine.cog as: spine_M_cog_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog as: spine_M_cog_ctrlSpace
building: char_bob.controls.spine.cog.cog as: spine_M_cog_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog as: spine_M_cog_ctrl
building: char_bob.controls.spine.cog.cog.spine01 as: spine_M_spine01_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.spine01 as: spine_M_spine01_ctrlSpace
building: char_bob.controls.spine.cog.cog.spine01.spine01 as: spine_M_spine01_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.spine01.spine01 as: spine_M_spine01_ctrl
building: char_bob.controls.spine.cog.cog.spine01.spine01.spine02 as: spine_M_spine02_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.spine01.spine01.spine02 as: spine_M_spine02_ctrlSpace
building: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02 as: spine_M_spine02_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02 as: spine_M_spine02_ctrl
building: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03 as: spine_M_spine03_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03 as: spine_M_spine03_ctrlSpace
building: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03.spine03 as: spine_M_spine03_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03.spine03 as: spine_M_spine03_ctrl
building: char_bob.controls.spine.cog.cog.spine04 as: spine_M_spine04_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.spine04 as: spine_M_spine04_ctrlSpace
building: char_bob.controls.spine.cog.cog.spine04.spine04 as: spine_M_spine04_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.spine04.spine04 as: spine_M_spine04_ctrl
building: char_bob.controls.spine.cog.cog.pelvis as: spine_M_pelvis_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.pelvis as: spine_M_pelvis_ctrlSpace
building: char_bob.controls.spine.cog.cog.pelvis.pelvis as: spine_M_pelvis_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.pelvis.pelvis as: spine_M_pelvis_ctrl
building: char_bob.controls.clavicleGuide as: clavicleGuide_L_cmp type: ComponentGroup
buildGroup: char_bob.controls.clavicleGuide as: clavicleGuide_L_cmp
building: char_bob.controls.clavicleGuide.inputs as: clavicleGuide_L_inputs_hrc type: HierarchyGroup
buildHierarchyGroup: char_bob.controls.clavicleGuide.inputs as: clavicleGuide_L_inputs_hrc
building: char_bob.controls.clavicleGuide.inputs.spineEnd as: clavicleGuide_L_spineEnd_cmpIn type: ComponentInput
buildLocator: char_bob.controls.clavicleGuide.inputs.spineEnd as: clavicleGuide_L_spineEnd_cmpIn
building: char_bob.controls.clavicleGuide.outputs as: clavicleGuide_L_outputs_hrc type: HierarchyGroup
buildHierarchyGroup: char_bob.controls.clavicleGuide.outputs as: clavicleGuide_L_outputs_hrc
building: char_bob.controls.clavicleGuide.outputs.clavicle as: clavicleGuide_L_clavicle_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.clavicleGuide.outputs.clavicle as: clavicleGuide_L_clavicle_cmpOut
building: char_bob.controls.clavicleGuide.outputs.clavicleEnd as: clavicleGuide_L_clavicleEnd_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.clavicleGuide.outputs.clavicleEnd as: clavicleGuide_L_clavicleEnd_cmpOut
building: char_bob.controls.clavicleGuide.clavicle as: clavicleGuide_L_clavicle_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.clavicleGuide.clavicle as: clavicleGuide_L_clavicle_ctrlSpace
building: char_bob.controls.clavicleGuide.clavicle.clavicle as: clavicleGuide_L_clavicle_ctrl type: Control
buildControl:char_bob.controls.clavicleGuide.clavicle.clavicle as: clavicleGuide_L_clavicle_ctrl
buildLayer: char_bob.controls as: char_bob_controls
buildLayer: char_bob.deformers as: char_bob_deformers
building: char_bob.controls.spine as: spine_M_cmp type: ComponentGroup
buildGroup: char_bob.controls.spine as: spine_M_cmp
building: char_bob.controls.spine.inputs as: spine_M_inputs_hrc type: HierarchyGroup
buildHierarchyGroup: char_bob.controls.spine.inputs as: spine_M_inputs_hrc
building: char_bob.controls.spine.inputs.mainSrt as: spine_M_mainSrt_cmpIn type: ComponentInput
buildLocator: char_bob.controls.spine.inputs.mainSrt as: spine_M_mainSrt_cmpIn
building: char_bob.controls.spine.outputs as: spine_M_outputs_hrc type: HierarchyGroup
buildHierarchyGroup: char_bob.controls.spine.outputs as: spine_M_outputs_hrc
building: char_bob.controls.spine.outputs.cog as: spine_M_cog_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.cog as: spine_M_cog_cmpOut
building: char_bob.controls.spine.outputs.spineBase as: spine_M_spineBase_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spineBase as: spine_M_spineBase_cmpOut
building: char_bob.controls.spine.outputs.pelvis as: spine_M_pelvis_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.pelvis as: spine_M_pelvis_cmpOut
building: char_bob.controls.spine.outputs.spineEnd as: spine_M_spineEnd_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spineEnd as: spine_M_spineEnd_cmpOut
building: char_bob.controls.spine.outputs.spine01 as: spine_M_spine01_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spine01 as: spine_M_spine01_cmpOut
building: char_bob.controls.spine.outputs.spine02 as: spine_M_spine02_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spine02 as: spine_M_spine02_cmpOut
building: char_bob.controls.spine.outputs.spine03 as: spine_M_spine03_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spine03 as: spine_M_spine03_cmpOut
building: char_bob.controls.spine.outputs.spine04 as: spine_M_spine04_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.spine.outputs.spine04 as: spine_M_spine04_cmpOut
building: char_bob.controls.spine.cog as: spine_M_cog_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog as: spine_M_cog_ctrlSpace
building: char_bob.controls.spine.cog.cog as: spine_M_cog_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog as: spine_M_cog_ctrl
building: char_bob.controls.spine.cog.cog.spine01 as: spine_M_spine01_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.spine01 as: spine_M_spine01_ctrlSpace
building: char_bob.controls.spine.cog.cog.spine01.spine01 as: spine_M_spine01_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.spine01.spine01 as: spine_M_spine01_ctrl
building: char_bob.controls.spine.cog.cog.spine01.spine01.spine02 as: spine_M_spine02_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.spine01.spine01.spine02 as: spine_M_spine02_ctrlSpace
building: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02 as: spine_M_spine02_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02 as: spine_M_spine02_ctrl
building: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03 as: spine_M_spine03_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03 as: spine_M_spine03_ctrlSpace
building: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03.spine03 as: spine_M_spine03_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03.spine03 as: spine_M_spine03_ctrl
building: char_bob.controls.spine.cog.cog.spine04 as: spine_M_spine04_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.spine04 as: spine_M_spine04_ctrlSpace
building: char_bob.controls.spine.cog.cog.spine04.spine04 as: spine_M_spine04_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.spine04.spine04 as: spine_M_spine04_ctrl
building: char_bob.controls.spine.cog.cog.pelvis as: spine_M_pelvis_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.spine.cog.cog.pelvis as: spine_M_pelvis_ctrlSpace
building: char_bob.controls.spine.cog.cog.pelvis.pelvis as: spine_M_pelvis_ctrl type: Control
buildControl:char_bob.controls.spine.cog.cog.pelvis.pelvis as: spine_M_pelvis_ctrl
building: char_bob.deformers.spine as: spine_M_cmp type: ComponentGroup
buildGroup: char_bob.deformers.spine as: spine_M_cmp
building: char_bob.deformers.spine.spine01 as: spine_M_spine01_def type: Joint
buildJoint: char_bob.deformers.spine.spine01 as: spine_M_spine01_def
building: char_bob.deformers.spine.pelvis as: spine_M_pelvis_def type: Joint
buildJoint: char_bob.deformers.spine.pelvis as: spine_M_pelvis_def
building: char_bob.deformers.spine.spine02 as: spine_M_spine02_def type: Joint
buildJoint: char_bob.deformers.spine.spine02 as: spine_M_spine02_def
building: char_bob.deformers.spine.spine03 as: spine_M_spine03_def type: Joint
buildJoint: char_bob.deformers.spine.spine03 as: spine_M_spine03_def
building: char_bob.deformers.spine.spine04 as: spine_M_spine04_def type: Joint
buildJoint: char_bob.deformers.spine.spine04 as: spine_M_spine04_def
building: char_bob.controls.clavicleGuide as: clavicleGuide_L_cmp type: ComponentGroup
buildGroup: char_bob.controls.clavicleGuide as: clavicleGuide_L_cmp
building: char_bob.controls.clavicleGuide.inputs as: clavicleGuide_L_inputs_hrc type: HierarchyGroup
buildHierarchyGroup: char_bob.controls.clavicleGuide.inputs as: clavicleGuide_L_inputs_hrc
building: char_bob.controls.clavicleGuide.inputs.spineEnd as: clavicleGuide_L_spineEnd_cmpIn type: ComponentInput
buildLocator: char_bob.controls.clavicleGuide.inputs.spineEnd as: clavicleGuide_L_spineEnd_cmpIn
building: char_bob.controls.clavicleGuide.outputs as: clavicleGuide_L_outputs_hrc type: HierarchyGroup
buildHierarchyGroup: char_bob.controls.clavicleGuide.outputs as: clavicleGuide_L_outputs_hrc
building: char_bob.controls.clavicleGuide.outputs.clavicle as: clavicleGuide_L_clavicle_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.clavicleGuide.outputs.clavicle as: clavicleGuide_L_clavicle_cmpOut
building: char_bob.controls.clavicleGuide.outputs.clavicleEnd as: clavicleGuide_L_clavicleEnd_cmpOut type: ComponentOutput
buildLocator: char_bob.controls.clavicleGuide.outputs.clavicleEnd as: clavicleGuide_L_clavicleEnd_cmpOut
building: char_bob.controls.clavicleGuide.clavicle as: clavicleGuide_L_clavicle_ctrlSpace type: CtrlSpace
buildGroup: char_bob.controls.clavicleGuide.clavicle as: clavicleGuide_L_clavicle_ctrlSpace
building: char_bob.controls.clavicleGuide.clavicle.clavicle as: clavicleGuide_L_clavicle_ctrl type: Control
buildControl:char_bob.controls.clavicleGuide.clavicle.clavicle as: clavicleGuide_L_clavicle_ctrl
building: char_bob.deformers.clavicleGuide as: clavicleGuide_L_cmp type: ComponentGroup
buildGroup: char_bob.deformers.clavicleGuide as: clavicleGuide_L_cmp
building: char_bob.deformers.clavicleGuide.clavicle as: clavicleGuide_L_clavicle_def type: Joint
buildJoint: char_bob.deformers.clavicleGuide.clavicle as: clavicleGuide_L_clavicle_def
connectAttribute: char_bob.controls.spine.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.inputs.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.inputs.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.inputs.inputs.drawDebug
connectAttribute: char_bob.controls.spine.inputs.inputs.rigScale
connectAttribute: char_bob.controls.spine.inputs.inputs.length
connectAttribute: char_bob.controls.spine.inputs.mainSrt.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.inputs.mainSrt.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.outputs.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.outputs.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.outputs.cog.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.outputs.cog.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.outputs.spineBase.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.outputs.spineBase.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.outputs.pelvis.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.outputs.pelvis.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.outputs.spineEnd.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.outputs.spineEnd.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.outputs.spine01.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.outputs.spine01.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.outputs.spine02.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.outputs.spine02.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.outputs.spine03.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.outputs.spine03.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.outputs.spine04.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.outputs.spine04.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03.spine03.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03.spine03.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.spine04.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.spine04.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.spine04.spine04.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.spine04.spine04.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.pelvis.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.pelvis.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.spine.cog.cog.pelvis.pelvis.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.spine.cog.cog.pelvis.pelvis.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.deformers.spine.implicitAttrGrp.visibility
connectAttribute: char_bob.deformers.spine.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.deformers.spine.spine01.implicitAttrGrp.visibility
connectAttribute: char_bob.deformers.spine.spine01.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.deformers.spine.pelvis.implicitAttrGrp.visibility
connectAttribute: char_bob.deformers.spine.pelvis.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.deformers.spine.spine02.implicitAttrGrp.visibility
connectAttribute: char_bob.deformers.spine.spine02.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.deformers.spine.spine03.implicitAttrGrp.visibility
connectAttribute: char_bob.deformers.spine.spine03.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.deformers.spine.spine04.implicitAttrGrp.visibility
connectAttribute: char_bob.deformers.spine.spine04.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.clavicleGuide.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.clavicleGuide.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.clavicleGuide.inputs.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.clavicleGuide.inputs.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.clavicleGuide.inputs.inputs.drawDebug
connectAttribute: char_bob.controls.clavicleGuide.inputs.inputs.rigScale
connectAttribute: char_bob.controls.clavicleGuide.inputs.inputs.rightSide
connectAttribute: char_bob.controls.clavicleGuide.inputs.spineEnd.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.clavicleGuide.inputs.spineEnd.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.clavicleGuide.outputs.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.clavicleGuide.outputs.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.clavicleGuide.outputs.clavicle.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.clavicleGuide.outputs.clavicle.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.clavicleGuide.outputs.clavicleEnd.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.clavicleGuide.outputs.clavicleEnd.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.clavicleGuide.clavicle.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.clavicleGuide.clavicle.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.controls.clavicleGuide.clavicle.clavicle.implicitAttrGrp.visibility
connectAttribute: char_bob.controls.clavicleGuide.clavicle.clavicle.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.deformers.clavicleGuide.implicitAttrGrp.visibility
connectAttribute: char_bob.deformers.clavicleGuide.implicitAttrGrp.shapeVisibility
connectAttribute: char_bob.deformers.clavicleGuide.clavicle.implicitAttrGrp.visibility
connectAttribute: char_bob.deformers.clavicleGuide.clavicle.implicitAttrGrp.shapeVisibility
buildConnection: spineEnd
buildXfoConnection: char_bob.clavicleGuide.spineEnd
buildPoseConstraint: char_bob.controls.clavicleGuide.inputs.spineEnd.spineEnd_To_spine03_PoseConstraint to: char_bob.controls.clavicleGuide.inputs.spineEnd
buildKLOperator:char_bob.spine.spineKLOp
buildKLOperator:char_bob.spine.spineDeformerKLOp
buildKLOperator:char_bob.spine.pelvisDeformerKLOp
buildPoseConstraint: char_bob.controls.spine.outputs.cog.cog_To_cog to: char_bob.controls.spine.outputs.cog
buildPoseConstraint: char_bob.controls.spine.outputs.spineBase.spineBase_To_spineBase to: char_bob.controls.spine.outputs.spineBase
buildPoseConstraint: char_bob.controls.spine.outputs.pelvis.pelvis_To_pelvis to: char_bob.controls.spine.outputs.pelvis
buildPoseConstraint: char_bob.controls.spine.outputs.spineEnd.spineEnd_To_spineEnd to: char_bob.controls.spine.outputs.spineEnd
buildPoseConstraint: char_bob.controls.spine.cog.cog_To_mainSrt to: char_bob.controls.spine.cog
buildKLOperator:char_bob.clavicleGuide.clavicleDeformerKLOp
buildPoseConstraint: char_bob.controls.clavicleGuide.inputs.spineEnd.spineEnd_To_spine03_PoseConstraint to: char_bob.controls.clavicleGuide.inputs.spineEnd
buildPoseConstraint: char_bob.controls.clavicleGuide.outputs.clavicle.clavicle_To_clavicle to: char_bob.controls.clavicleGuide.outputs.clavicle
buildPoseConstraint: char_bob.controls.clavicleGuide.outputs.clavicleEnd.clavicleEnd_To_clavicle to: char_bob.controls.clavicleGuide.outputs.clavicleEnd
buildPoseConstraint: char_bob.controls.clavicleGuide.clavicle.clavicle_To_spineEnd to: char_bob.controls.clavicleGuide.clavicle
char_bob
char_bob.spine:M
char_bob.controls
char_bob.controls.spine:M
char_bob.controls.spine:M.inputs
char_bob.controls.spine:M.inputs.mainSrt
char_bob.controls.spine:M.outputs
char_bob.controls.spine:M.outputs.cog
char_bob.controls.spine:M.outputs.spineBase
char_bob.controls.spine:M.outputs.pelvis
char_bob.controls.spine:M.outputs.spineEnd
char_bob.controls.spine:M.outputs.spine01
char_bob.controls.spine:M.outputs.spine02
char_bob.controls.spine:M.outputs.spine03
char_bob.controls.spine:M.outputs.spine04
char_bob.controls.spine:M.cog
char_bob.controls.spine:M.cog.cog
char_bob.controls.spine:M.cog.cog.spine01
char_bob.controls.spine:M.cog.cog.spine01.spine01
char_bob.controls.spine:M.cog.cog.spine01.spine01.spine02
char_bob.controls.spine:M.cog.cog.spine01.spine01.spine02.spine02
char_bob.controls.spine:M.cog.cog.spine01.spine01.spine02.spine02.spine03
char_bob.controls.spine:M.cog.cog.spine01.spine01.spine02.spine02.spine03.spine03
char_bob.controls.spine:M.cog.cog.spine04
char_bob.controls.spine:M.cog.cog.spine04.spine04
char_bob.controls.spine:M.cog.cog.pelvis
char_bob.controls.spine:M.cog.cog.pelvis.pelvis
char_bob.controls.clavicleGuide:L
char_bob.controls.clavicleGuide:L.inputs
char_bob.controls.clavicleGuide:L.inputs.spineEnd
char_bob.controls.clavicleGuide:L.outputs
char_bob.controls.clavicleGuide:L.outputs.clavicle
char_bob.controls.clavicleGuide:L.outputs.clavicleEnd
char_bob.controls.clavicleGuide:L.clavicle
char_bob.controls.clavicleGuide:L.clavicle.clavicle
char_bob.deformers
char_bob.deformers.spine:M
char_bob.deformers.spine:M.spine01
char_bob.deformers.spine:M.pelvis
char_bob.deformers.spine:M.spine02
char_bob.deformers.spine:M.spine03
char_bob.deformers.spine:M.spine04
char_bob.deformers.clavicleGuide:L
char_bob.deformers.clavicleGuide:L.clavicle
char_bob.clavicleGuide:L
